src/components/QuantityInput/QuantityInput.js

```diff
@@ -66,41 +66,61 @@ const QuantityInput = ({
   maxQuantity,
   setQuantity,
   value,
-}) => (
-  <div className="QuantityInput">
-    <QuantityTextInput
-      {...{ handleSubmit, handleUpdateNumber, maxQuantity, value }}
-    />
-    <span className="quantity">
-      /{' '}
-      <AnimatedNumber {...{ number: maxQuantity, formatter: integerString }} />
-    </span>
-    <div className="number-nudger-container">
-      <Fab
-        disabled={!value}
-        {...{
-          'aria-label': 'Increment',
-          color: 'primary',
-          onClick: () => setQuantity(++value > maxQuantity ? 1 : value),
```

Comment:
I just realized how poorly I initially implemented this. These event handlers are mutating `value`, so it's no wonder we're experiencing broken behavior. I have no idea why I though this would be a good idea. 🤦‍♂️

Reply:
all good! kinda interesting one honestly. i've always kind of avoided this syntax cause it's less common and tricky to read, but i've never seen it actually cause a bug until now

Reply:
It's a remnant from my C++ days in college. I should know better by now! 😂